File: src/main/java/org/chousik/collection/builder/collectors/StringCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.validators.IValidator;
import org.chousik.exception.InvalidDataError;
import org.chousik.exception.ScriptExecutionError;
import org.chousik.handlers.RunHandler;

import java.util.NoSuchElementException;
import java.util.Scanner;

public class StringCollector implements ICollector<String>{
    private Boolean isScript;
    private Scanner scanner;
    public void NumberCollecto() {
        this.isScript = RunHandler.mode();
        this.scanner = RunHandler.getMainScaner();
    }

    @Override
    public String get(String name, IValidator validator) throws ScriptExecutionError {
        while (true) {
            try {
                if (!isScript) {
                    System.out.println("Введите " + name);
                }
                String namePerson = scanner.nextLine().trim();
                validator.valide(namePerson);
                return namePerson;
            } catch (InvalidDataError e) {
                if (isScript) {
                    throw new ScriptExecutionError("Поле " + name + " не может быть пустым.");
                }
                System.out.println("Поле " + name + " не может быть пустым.");
            } catch (NullPointerException e) {
                if (isScript) {
                    throw new ScriptExecutionError("Поле " + name + " не может быть null.");
                }
                System.out.println("Поле " + name + " не может быть null.");
            } catch (NoSuchElementException e) {
                if (isScript) {
                    throw new ScriptExecutionError("Ошибка во время ввода данных коллекции из файла. Конец файла.");
                }
                System.out.println("Не нажимай Ctrl+D((((");
                System.exit(0);
            } catch (Exception e) {
                if (isScript) {
                    throw new ScriptExecutionError("Непридвиденная ошибка");
                }
                System.out.println("Непридвиденная ошибка");
                System.exit(0);
            }
        }
    }
}

File: src/main/kotlin/collection/builder/collectors/StringCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.validators.IValidator
import exeption.InvalidDataError
import exeption.ScriptExecutionError
import org.chousik.handlers.RunHandler
import java.util.*
import kotlin.system.exitProcess

class StringCollector : ICollector<String, String?> {
    private val isScript = RunHandler.mode()
    private val scanner: Scanner = RunHandler.getMainScanner()

    override fun ask(name: String, validator: IValidator<String?>): String {
        while (true) {
            try {
                if (!isScript) {
                    println("Введите $name")
                    print(System.getProperty("user.name") + "> ")
                }
                val namePerson = scanner.nextLine().trim { it <= ' ' }
                validator.valid(namePerson)
                return namePerson
            } catch (e: InvalidDataError) {
                if (isScript) {
                    throw ScriptExecutionError("Поле $name не может быть пустым.")
                }
                println("Поле $name не может быть пустым.")
            } catch (e: NullPointerException) {
                if (isScript) {
                    throw ScriptExecutionError("Поле $name не может быть null.")
                }
                println("Поле $name не может быть null.")
            } catch (e: NoSuchElementException) {
                if (isScript) {
                    throw ScriptExecutionError("Ошибка во время ввода данных коллекции из файла. Конец файла.")
                }
                println("Не нажимай Ctrl+D((((")
                exitProcess(0)
            } catch (e: Exception) {
                if (isScript) {
                    throw ScriptExecutionError("Непредвиденная ошибка")
                }
                println("Непредвиденная ошибка")
                exitProcess(0)
            }
        }
    }
}